cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/spec/StageSpec.java

```diff
@@ -315,4 +315,11 @@ public int hashCode() {
       return Objects.hash(port, schema);
     }
   }
+
+  public static StageSpec createCopyWithUpdatedMaxPreviewRecords(StageSpec stageSpec, int newMaxPreviewRecords) {
+    return new StageSpec(stageSpec.name, stageSpec.plugin, stageSpec.inputSchemas, stageSpec.outputSchema,
+                         stageSpec.errorSchema, stageSpec.portSchemas, stageSpec.outputPorts,
+                         stageSpec.stageLoggingEnabled, stageSpec.processTimingEnabled,
+                         Math.min(stageSpec.maxPreviewRecords, newMaxPreviewRecords));
```

Comment:
better to LOG warning if are overriding the limit provided by user through pipeline config i.e. when  stageSpec has maxPreviewRecords greater than limit from cdap-site config